lambda_uploader/config.py

```diff
@@ -78,20 +81,39 @@ def _validate(self, key, cls=None):
         if key not in self._config:
             raise ValueError("Config %s must have %s set"
                              % (self._path, key))
-            if cls:
-                if not isinstance(self._config[key], cls):
-                    raise TypeError("Config value '%s' should be %s not %s"
-                                    % (key, cls, type(self._config[key])))
+
+            return self._compare(key, cls, self._config[key])
+
+    '''Validate the VPC configuration'''
+    def _validate_vpc(self):
+        for param, clss in REQUIRED_VPC_PARAMS.iteritems():
+            self._compare(param, clss, self._config['vpc'].get(param))
+
+            if len(self._config['vpc'].get(param)) == 0:
+                raise TypeError("VPC Config '%s' should have at least"
+                                " one item in its array!" % param)
+            for value in self._config['vpc'].get(param):
+                if not isinstance(value, basestring):
+                    raise TypeError("VPC Config arrays can only contain"
+                                    " strings. '%s' contains something else"
+                                    % param)
+
+    '''Compare if a string is a certain type'''
+    def _compare(self, key, cls, value):
+        if cls:
+            if not isinstance(value, cls):
+                raise TypeError("Config value '%s' should be %s not %s"
+                                % (key, cls, type(value)))
 
     '''Load config ... called by init()'''
     def _load_config(self, lambda_file=None):
         if not lambda_file:
             lambda_file = path.join(self._path, 'lambda.json')
 
-        if not path.isfile(lambda_file):
+        if not path.isfile(path.join(self._path, lambda_file)):
```

Comment:
@martinb3 I totally thought it was a bug! I was expecting that you could just pass a different name for a config in the directory, but I guess it gives you more options (like one `lambda.json` for multiple projects). Updated to drop this change.